src/compute-model-statistics/src/main/scala/ComputeModelStatistics.scala

```diff
@@ -245,13 +246,8 @@ class ComputeModelStatistics(override val uid: String) extends Transformer with
                                          confusionMatrix: Matrix,
                                          resultDF: DataFrame): DataFrame = {
     var resultDFModified = resultDF
-    for (col: Int <- 0 until confusionMatrix.numCols;
-         row: Int <- 0 until confusionMatrix.numRows) {
-      resultDFModified = resultDFModified
-        .withColumn(s"predicted_class_as_${labels(col).toString}_actual_is_${labels(row).toString}",
-          lit(confusionMatrix(row, col)))
-    }
-    resultDFModified
+    val schema = resultDFModified.schema.add(MetricConstants.ConfusionMatrix, SQLDataTypes.MatrixType)
```

Comment:
You should make the output column configurable with HasOutputCol, and give this as a default

Reply:
discussed this